flink-runtime/src/main/java/org/apache/flink/runtime/security/SecurityContext.java

```diff
@@ -155,6 +157,58 @@ public static void install(SecurityConfiguration config) throws Exception {
 		installedContext = new SecurityContext(loginUser);
 	}
 
+	/*
+	 * This is a temporary fix to support both Kafka and ZK client libraries
+	 * that are expecting the system variable to determine secure cluster
+	 */
+	private static void populateJaasConfigSystemProperty(Configuration configuration) {
+
+		//hack since Kafka Login Handler explicitly looks for the property or else it throws an exception
+		//https://github.com/apache/kafka/blob/0.9.0/clients/src/main/java/org/apache/kafka/common/security/kerberos/Login.java#L289
+		if(null == configuration) {
+			System.setProperty("java.security.auth.login.config", "");
+			return;
+		}
+
+		String baseDir = configuration.getString(ConfigConstants.FLINK_BASE_DIR_PATH_KEY, null);
+		if(baseDir == null) {
+			System.setProperty("java.security.auth.login.config", "");
+			return;
+		}
+
+		File f = new File(baseDir);
+		if(!f.exists() || !f.isDirectory()) {
+			LOG.error("Invalid flink base directory {} configuration provided", baseDir);
+			throw new IllegalConfigurationException("Invalid flink base directory configuration provided");
+		}
```

Comment:
This check should really be somewhere else. I think you can use `ENV_FLINK_CONF_DIR` which is always set. The `FLINK_BASE_DIR_PATH_KEY` should be deprecated. It's not used anymore.